x-pack/plugins/cases/server/saved_object_types/migrations/cases.test.ts

```diff
@@ -0,0 +1,351 @@
+import { caseConnectorIdMigration } from './cases';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const create_7_14_0_case = ({
+  connector,
+  externalService,
+}: { connector?: ESCaseConnectorWithId; externalService?: CaseFullExternalService } = {}) => ({
+  type: CASE_SAVED_OBJECT,
+  id: '1',
+  attributes: {
+    connector,
+    external_service: externalService,
+  },
+});
+
+describe('7.15.0 connector ID migration', () => {
+  it('does not create a reference when the connector.id is none', () => {
+    const caseSavedObject = create_7_14_0_case({ connector: getNoneCaseConnector() });
+
+    const migratedConnector = caseConnectorIdMigration(
+      caseSavedObject
+    ) as SavedObjectSanitizedDoc<CaseAttributes>;
+
+    expect(migratedConnector.references.length).toBe(0);
+    expect(migratedConnector.attributes.connector).not.toHaveProperty('id');
+    expect(migratedConnector.attributes.connector).toMatchInlineSnapshot(`
+      Object {
+        "fields": null,
+        "name": "none",
+        "type": ".none",
+      }
+    `);
+  });
+
+  it('does not create a reference when the connector is undefined', () => {
+    const caseSavedObject = create_7_14_0_case();
+
+    const migratedConnector = caseConnectorIdMigration(
+      caseSavedObject
+    ) as SavedObjectSanitizedDoc<CaseAttributes>;
+
+    expect(migratedConnector.references.length).toBe(0);
+    expect(migratedConnector.attributes.connector).not.toHaveProperty('id');
+    expect(migratedConnector.attributes.connector).toMatchInlineSnapshot(`
+      Object {
+        "fields": null,
+        "name": "none",
+        "type": ".none",
+      }
+    `);
+  });
+
+  it('sets the connector to the default none connector if the connector.id is undefined', () => {
+    const caseSavedObject = create_7_14_0_case({
+      connector: {
+        fields: null,
+        name: ConnectorTypes.jira,
+        type: ConnectorTypes.jira,
+      } as ESCaseConnectorWithId,
+    });
+
+    const migratedConnector = caseConnectorIdMigration(
+      caseSavedObject
+    ) as SavedObjectSanitizedDoc<CaseAttributes>;
+
+    expect(migratedConnector.references.length).toBe(0);
+    expect(migratedConnector.attributes.connector).not.toHaveProperty('id');
+    expect(migratedConnector.attributes.connector).toMatchInlineSnapshot(`
+      Object {
+        "fields": null,
+        "name": "none",
+        "type": ".none",
+      }
+    `);
+  });
+
+  it('does not create a reference when the external_service is null', () => {
+    const caseSavedObject = create_7_14_0_case({ externalService: null });
+
+    const migratedConnector = caseConnectorIdMigration(
+      caseSavedObject
+    ) as SavedObjectSanitizedDoc<CaseAttributes>;
+
+    expect(migratedConnector.references.length).toBe(0);
+    expect(migratedConnector.attributes.external_service).toBeNull();
+  });
+
+  it('does not create a reference when the external_service is undefined', () => {
```

Comment:
Based on https://github.com/elastic/kibana/pull/104221/files?file-filters%5B%5D=.gz&file-filters%5B%5D=.ts#r680887363 I think we should check that if the external_service is undefined then `expect(migratedConnector.attributes.external_service).toBeNull();`.